--- e2fsprogs-1.42.12/misc/e4defrag.c.omv~	2014-08-30 23:40:39.969612912 +0200
+++ e2fsprogs-1.42.12/misc/e4defrag.c	2014-08-30 23:42:42.259084083 +0200
@@ -198,7 +198,7 @@ static struct frag_statistic_ino	frag_ra
 #endif /* ! HAVE_SYNC_FILE_RANGE */
 
 #ifndef HAVE_FALLOCATE64
-#error fallocate64 not available!
+#define fallocate64(a,b,c,d) posix_fallocate64(a,c,d)
 #endif /* ! HAVE_FALLOCATE64 */
 
 /*
